src/ui/marker.js

```diff
@@ -11,27 +11,43 @@ import type Popup from './popup';
 import type {LngLatLike} from "../geo/lng_lat";
 import type {MapMouseEvent} from './events';
 
+export type Anchor = 'middle';
```

Comment:
You had it right the first time enumerating the valid options like https://github.com/mapbox/mapbox-gl-js/blob/master/src/ui/popup.js#L19

@anandthakker any thoughts on if Marker should import the Anchor type from Popup? That would mean adding `middle` as a popup anchor option...